### PR TITLE
DNM Modify nameservers and search domains in place

### PR DESCRIPTION
**Pull request by karthiksundaravel**
Added the ability to modify DNS servers as well as DNS search domains on the fly. Also cleaned up code and updated much of the string handling for Python3.

Jira: https://issues.redhat.com/browse/OSPRH-10528
BZ: https://bugzilla.redhat.com/show_bug.cgi?id=2315341

### CODE REVIEW

**Review by vcandapp — CHANGES_REQUESTED** [review]

**Review by dsneddon — APPROVED** [review]

File: os_net_config/impl_ifcfg.py [hunk 1]
```diff
@@ -306,38 +305,58 @@ def iproute2_apply_commands(self, device_name, filename, data):
...
+            if changes['IPADDR'] == 'added' or changes['IPADDR'] == 'modified':
+                ip = data_values['IPADDR']
+                commands.insert(0, f'addr add {ip}/{new_cidr} dev {dev_name}')
+        if 'MTU' in changes:
+            if changes['MTU'] == 'added' or changes['MTU'] == 'modified':
+                mtu = data_values['MTU']
+                commands.append(f'link set dev {dev_name} mtu {mtu}')
+            elif changes['MTU'] == 'removed':
+                commands.append(f'link set dev {dev_name} mtu 1500')
+        if 'DOMAIN' in changes:
```
[the remark this answers is not in the document]
Comment (dsneddon) [reply os_net_config/impl_ifcfg.py:339]: Viji,

This file is going to be overwritten every time the interface is brought up. When the user reboots this file will be overwritten, so there is no reason to add the comment that you requested.

**Review by vcandapp — APPROVED** [review]
lgtm

File: os_net_config/impl_ifcfg.py [hunk 2]
```diff
@@ -306,38 +305,58 @@ def iproute2_apply_commands(self, device_name, filename, data):
...
+            old = file_values['DOMAIN']
+            new = data_values['DOMAIN']
```
Comment (vcandapp) [inline os_net_config/impl_ifcfg.py:341]: Domain should be like https://github.com/os-net-config/os-net-config/blob/master/os_net_config/impl_ifcfg.py#L759

**Comment by karthiksundaravel** [comment]
Abandonned

**Pull request closed without merging**